db/migrate/20180503120921_create_calendar_events.rb

```diff
@@ -0,0 +1,14 @@
+class CreateCalendarEvents < ActiveRecord::Migration[5.2]
+  def change
+    create_table :calendar_events do |t|
+      t.references :academic_calendar, foreign_key: true
+      t.references :calendar_title, foreign_key: true
+      t.datetime :start_date, null: false
+      t.datetime :end_date
+
+      t.timestamps
+    end
+    add_index :calendar_events, [:academic_calendar_id, :calendar_title_id],
```

Comment:
Yukarıdaki sebeple buna da gerek var mı emin değilim, bakmak lazım.

Reply:
nested yapılarda uniqueness validasyonun scope ile kullanımı doğru çalışmıyor. Önceden bahsettiğim problem buydu. Ondan dolayı db bazlı validation ekledim. [ilgili sorun](https://github.com/rails/rails/issues/20676)

Reply:
Bu index oluşturma olaylarında dikkatli olmak gerekiyor. Bende çok hakim değilim fakat şimdi bir kaç dokümana  baktım. Postgresql için baktığımda yapılan sorgulamalara göre kullanılacak index türü değişiyor.(Not: Türler arasındaki fark kullandıkları sorgulama algoritmalarından kaynaklı) Eğer tür belirtilmezse index  `B-tree` türünde oluşturuluyor. Bir diğer hususta multi-column index ve single column index arasındaki kullanım farkı bununla ilgili de sorulara filan bakıyorum. Genel olarak sistemde sıklıkla yapılacak sorgulamalara göre oluşturulması gereken indexler değişiklik gösteriyor. 
Tabi bu database sorgu optimizasyon konusu ve halihazırda uygulamada veri olmadan, uygulama kullanılmadan bu tür optimizasyon hamleleri yapmak biraz zor, çünkü uygulama geliştikçe sıklıkla yapılan sorgularda değişiklik gösteriyor.

- [Postgresql Index Türleri](https://www.postgresql.org/docs/10/static/indexes-types.html)